src/main/java/Problem1.java

```diff
@@ -2,7 +2,36 @@
 
     // Do not change signature (function name, parameters)
     public static int binarySearch(int[] data, int target) {
-        // Add your solution
+        //Initializes the ranges of the array
+        int middleIndex = data.length / 2;
+        int low = 0;
+        int high = data.length - 1;
+
+        //Performs checks to qualify the array before it enters the for loop.
+        //If the following conditions are met, these preset values are returned
+        if (data == null || data.length == 0) {
+            return -1;
+        }
+        if (data.length == 1 && target != data[0]) {
+            return -1;
+        }
+        if (target == data[0]) {
+            return 0;
+        }
+
+        //Checks the middle number of the array to see if the requested number matches.
+        //Performs binary search until number is found.
+        for (int i = 0; i < data.length; i++) {
```

Comment:
-5 within the loop, it's linear search instead of binary search. for example, if the input is 

[1,2,3,4,5,6,7,8] and search 8

middleIndex will be 4, 5, 6, 7, 8

whereas it should be 4, 6, 7, 8